include/boost/url/detail/any_segments_iter.hpp

```diff
@@ -44,7 +44,9 @@ struct BOOST_SYMBOL_VISIBLE
     // 0 = zero
     // 1 = one
     // 2 = two, or more
-    std::size_t fast_nseg = 0;
+    int fast_nseg = 0;
+
+    bool encode_colons = false;
```

Comment:
well the way you pass it in the ctor, it looks like it doesn't need to be public. I was fine with just letting `edit_segments` set the data member directly, because it is less code (no need to repeat all the boilerplate in the derived iterator constructors). What do you want to do?

Reply:
Other functions depend on accessing both fast_nseg and encode_colons directly, like all other members. `edit_segments` depends on a lot of information to determine if encode_colons should be true.

Reply:
I was wrong... I misread. It wasn't the ctor it was the copy_impl and measure_impl.

Reply:
rename to `escape_colons` and in url_base.ipp please :)

Reply:
or not... but at least put a comment on the data member.

Reply:
LOL...